### DISCUSSION TITLE
Runtime: nodejs vs edge

### DISCUSSION THREAD

**Discussion by loeck** [vercel/next.js#69486] [Help] [answered]
### Summary

Hello!

We have been using Next.js for a while now on Vercel. We have a v1 with the "pages router" running on the Node.js runtime (for both rendering and API), and a v2 with the "app router" running on the Edge runtime (for both rendering and API).

I recall experiencing some cold start issues with the Node.js runtime, but it seems to have improved significantly.

In your opinion, which option should I choose moving forward?

Thank you!

### Additional information

Like I said, it's probably specific to Vercel, so I'm not sure if it's the right place to ask

**Comment by Juneezee** [comment, marked as answer]
The docs has a page that compares Node.js and Edge runtime, see https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes.

Basically if you need to use native Node.js modules (e.g. `fs.readFile`), you need to use Node.js runtime. Otherwise you can choose to use Edge Runtime if you are deploying to Vercel and would like to use their Edge Network.

You can read more about Vercel Edge here:

1. https://vercel.com/docs/functions/runtimes/edge-runtime
2. https://vercel.com/docs/edge-network/overview

**Reply by loeck** [reply]
Ok perfect, thanks for that !
